aten/src/ATen/core/ATenDispatch.h

```diff
@@ -44,70 +46,54 @@ class CAFFE2_API ATenOpTable {
-        "Attempting to register variable function for schema ", schema_,
-        " but there is already a function registered");
-    variable_function_ = fn;
-  }
-
-  void* getBaseOp(Backend backend) const {
-    if (function_table_[static_cast<int64_t>(backend)] == nullptr) {
-      TORCH_CHECK(function_table_[static_cast<int64_t>(Backend::Undefined)] != nullptr,
-          "No function is registered for schema ", schema_, " on backend ", toString(backend));
-      return function_table_[static_cast<int64_t>(Backend::Undefined)];
+  void* getOp(TensorTypeId tid) const {
+    // You might think we can minorly optimize this further by maintaining a
+    // bitmask of registered operator keys, so we don't select dispatch ids
+    // which don't have implementations here.  But the net effect is that if you
+    // get a Variable CPUTensor, if there is no variable registration, you'll
+    // fall back to the CPU implementation.  Is this what you want?  Unlikely...
+    if (function_table_[static_cast<int64_t>(tid)] == nullptr) {
+      TORCH_CHECK(function_table_[static_cast<int64_t>(TensorTypeId::UndefinedTensorId)] != nullptr,
+          "No function is registered for schema ", schema_, " on tensor type ", toString(tid));
+      return function_table_[static_cast<int64_t>(TensorTypeId::UndefinedTensorId)];
     }
-    return function_table_[static_cast<int64_t>(backend)];
-  }
-
-  void* getVariableOp() const {
-    TORCH_CHECK(variable_function_ != nullptr,
-        "No variable function registered for ", schema_);
-    return variable_function_;
+    return function_table_[static_cast<int64_t>(tid)];
   }
 
   friend class ATenDispatch;
 
   std::string schema_;
-  void* function_table_[static_cast<int64_t>(Backend::NumOptions)] = {nullptr};
+  void* function_table_[static_cast<int64_t>(TensorTypeId::NumTensorIds)] = {nullptr};
   void* variable_function_ = nullptr;
 };
 
 class CAFFE2_API ATenDispatch {
  public:
   template<class FuncType>
-  ATenDispatch& registerOp(Backend backend, const char* schema, FuncType* fn) {
+  ATenDispatch& registerOp(TensorTypeId id, const char* schema, FuncType* fn) {
     std::lock_guard<std::mutex> lock(mutex_);
     if (op_tables_.find(schema) == op_tables_.end()) {
       op_tables_.insert(std::make_pair(schema, ATenOpTable(schema)));
     }
-    op_tables_.at(schema).registerOp(backend, reinterpret_cast<void*>(fn));
+    op_tables_.at(schema).registerOp(id, reinterpret_cast<void*>(fn));
     return *this;
   }
 
-  template <class FuncType>
-  ATenDispatch& registerVariableOp(const char* schema, FuncType* fn) {
-    std::lock_guard<std::mutex> lock(mutex_);
-    if (op_tables_.find(schema) == op_tables_.end()) {
-      op_tables_.insert(std::make_pair(schema, ATenOpTable(schema)));
-    }
-    op_tables_.at(schema).registerVariableOp(reinterpret_cast<void*>(fn));
-    return *this;
+  template<class FuncType>
+  ATenDispatch& registerOp(Backend b, const char* schema, FuncType* fn) {
```

Comment:
how much is this still used? :).

Reply:
I don't know. Will have to remove and see, I guess!

Reply:
Answer: `aten/src/ATen/test/extension_backend_test.cpp`